src/main/scala/com/cognite/sdk/scala/common/login.scala

```diff
@@ -12,14 +14,23 @@ class Login[F[_]](val requestSession: RequestSession[F]) {
   implicit val loginStatusDecoder = deriveDecoder[LoginStatus]
   implicit val dataLoginStatusDecoder = deriveDecoder[DataLoginStatus]
   def status(): F[LoginStatus] =
-    requestSession
-      .sendCdf { request =>
-        request
-          .get(uri"${requestSession.baseUri}/login/status")
-          .response(asJson[DataLoginStatus])
-          .mapResponse {
-            case Left(value) => throw value.error
-            case Right(value) => value.data
-          }
-      }
+    try {
+      requestSession
+        .sendCdf { request =>
+          request
+            .get(uri"${requestSession.baseUri}/login/status")
+            .response(asJson[DataLoginStatus])
+            .mapResponse {
+              case Left(value) => throw value.error
+              case Right(value) => value.data
+            }
+        }
+    } catch {
+      case unknownHost: UnknownHostException => throw unknownHost
+      case connectionTimeout: ConnectException => throw connectionTimeout
+      case _: Throwable =>
```

Comment:
We shouldn't catch `Throwable`, be more specific. If we can't be more specific in a sensible way, it's better to not catch.